library/core/src/mem/mod.rs

```diff
@@ -1178,3 +1178,18 @@ pub const fn discriminant<T>(v: &T) -> Discriminant<T> {
 pub const fn variant_count<T>() -> usize {
     intrinsics::variant_count::<T>()
 }
+
+/// This is here only to simplify all the ZST checks we need in the library.
+/// It's not on a stabilization track right now.
```

Comment:
Can you add some more documentation for using this? Maybe a small example? E.g. I'd like to be able to just say "this should use `core::mem::SizedTypeProperties::IS_ZST`" in a review, without having to explain much more.